src/legacy/core_plugins/kibana/common/field_formats/types/url.ts

```diff
@@ -0,0 +1,191 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { escape, memoize } from 'lodash';
+import {
+  getHighlightHtml,
+  FieldFormat,
+  TEXT_CONTEXT_TYPE,
+  HTML_CONTEXT_TYPE,
+} from '../../../../../../plugins/data/common/';
+
+const templateMatchRE = /{{([\s\S]+?)}}/g;
+const whitelistUrlSchemes = ['http://', 'https://'];
+
+const URL_TYPES = [
+  { kind: 'a', text: 'Link' },
+  { kind: 'img', text: 'Image' },
+  { kind: 'audio', text: 'Audio' },
+];
+const DEFAULT_URL_TYPE = 'a';
+
+export function createUrlFormat() {
+  class UrlFormat extends FieldFormat {
+    static id = 'url';
+    static title = 'Url';
+    static fieldType = [
+      'number',
+      'boolean',
+      'date',
+      'ip',
+      'string',
+      'murmur3',
+      'unknown',
+      'conflict',
+    ];
+    static urlTypes = URL_TYPES;
+
+    constructor(params: any) {
+      super(params);
+      this.compileTemplate = memoize(this.compileTemplate);
+    }
+
+    getParamDefaults() {
+      return {
+        type: DEFAULT_URL_TYPE,
+        urlTemplate: null,
+        labelTemplate: null,
+      };
+    }
+
+    private formatLabel(value: any, url?: any) {
+      const template = this.param('labelTemplate');
+      if (url == null) url = this.formatUrl(value);
+      if (!template) return url;
+
+      return this.compileTemplate(template)({
+        value,
+        url,
+      });
+    }
+
+    private formatUrl(value: any) {
+      const template = this.param('urlTemplate');
+      if (!template) return value;
+
+      return this.compileTemplate(template)({
+        value: encodeURIComponent(value),
+        rawValue: value,
+      });
+    }
+
+    private compileTemplate(template: string) {
+      const parts = template.split(templateMatchRE).map(function(part: string, i: number) {
+        // trim all the odd bits, the variable names
+        return i % 2 ? part.trim() : part;
+      });
+
+      return function(locals: Record<any, any>) {
+        // replace all the odd bits with their local var
+        let output = '';
+        let i = -1;
+        while (++i < parts.length) {
+          if (i % 2) {
+            if (locals.hasOwnProperty(parts[i])) {
+              const local = locals[parts[i]];
+              output += local == null ? '' : local;
+            }
+          } else {
+            output += parts[i];
+          }
+        }
+
+        return output;
+      };
+    }
+
+    _convert = {
+      [TEXT_CONTEXT_TYPE](this: UrlFormat, value: any): string {
+        return this.formatLabel(value);
+      },
+
+      [HTML_CONTEXT_TYPE](
+        this: UrlFormat,
+        rawValue: any,
+        field: { name: string | number },
+        hit: { highlight: { [x: string]: any } },
```

Comment:
We already have types for that. What about: 
![image](https://user-images.githubusercontent.com/20072247/66058811-aed0cd80-e543-11e9-9970-5e94a3fbeed1.png)

Reply:
Nice catch! Replaced everywhere.